lib/rails_cursor_pagination/paginator.rb

```diff
@@ -143,19 +147,21 @@ def page_info
     #
     # @return [Array<Hash>] List of hashes, each with a `cursor` and `data`
     def page
-      records.map do |item|
-        {
-          cursor: cursor_for_record(item),
-          data: item
-        }
+      memoize :page do
+        records.map do |item|
+          {
+            cursor: cursor_for_record(item),
+            data: item
+          }
+        end
       end
     end
 
     # Get the total number of records in the given relation
     #
     # @return [Integer]
     def total
-      memoize(:total) { @relation.size }
+      memoize(:total) { @relation.reorder('').size }
```

Comment:
why do we need to reorder in this case, isn't the result of `size` independent from the order?